YoCode/TestCountCheck.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace YoCode
+{
+    public class TestCountCheck
+    {
+        string processName;
+        string workingDir;
+        string arguments;
+
+        public string statLine { get; set; }
+        public string Output { get; set; }
+
+        TestStats stats;
+        List<int> tempStats;
+
+
+        public TestCountCheck(string repositoryPath)
+        {
+            processName = "dotnet";
+            workingDir = repositoryPath;
+            arguments = "test";
+        }
+
+        public void ExecuteTheCheck()
+        {
+            ProcessRunner pr = new ProcessRunner(processName, workingDir, arguments);
+            pr.ExecuteTheCheck();
+           
+            Output = pr.Output;
+            statLine = Output.GetLineWithAllKeywords(GetTestKeyWords());
+            tempStats = CountNumberOfTests(statLine);
+            StoreCalculations(tempStats); 
+
+        }
+
+        public List<int> CountNumberOfTests(String statLine)
+        {
+            string[] numbers = Regex.Split(statLine, @"\D+");
```

Comment:
It might be worth having a more specific regex here to make sure the numbers on the stats line really are in the order you think they are. My main reason for saying that is that it could break without us realising. If the order of the output on the line was different than we expect, it could still appear to be working even though it's giving the wrong results. It's also a great candidate for a unit test. I would add at least one unit test to check the behaviour and then once the test is passing, try to make a more precise regex.